### PR TITLE
Api gateway plugin and and interactive CLI

### PR DESCRIPTION
**Pull request by AlexisNo**

### CODE REVIEW

**Comment by AlexisNo** [comment]
@cs-heucles if you want to have a look before merging ;)

File: bin/lager [hunk 1]
```diff
@@ -6,8 +6,5 @@ const program = lager.getProgram();
 
 program
   .version('0.0.1')
```
Comment (cs-heucles) [inline bin/lager]: For now all of this commands will not exist anymore?

Comment (AlexisNo) [inline bin/lager]: They are implemented in their respective plugin. in this case, the `api-gateway` pĺugin.
A Lager plugin can add new commands to the Lager cli.

File: plugins/api-gateway/bin/inspect-api.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+'use strict';
```
Comment (AlexisNo) [inline plugins/api-gateway/bin/inspect-api.js]: @cs-heucles The `inspect-api` subcommand is part of the `api-gateway` plugin.
In fact the Lager core only does 2 things:
- Provide a promise based Event/Observer system for plugin's hooks
- Provide a base CLI that can be extended by plugins

**Pull request merged**